hm_1/tasks/task_07_01.py

```diff
@@ -0,0 +1,46 @@
+"""
+Given a dictionary (tree), that can contains multiple nested structures.
+Write a function, that takes element and finds the number of occurrences
+of this element in the tree.
+Tree can only contains basic structures like:
+    str, list, tuple, dict, set, int, bool
+"""
+from typing import Any, List
+
+
+def iterate_all(iterable, returned="key"):
+    if isinstance(iterable, dict):
+
+        for key, value in iterable.items():
+
+            if returned == "key":
+                yield [key, value]
+            elif returned == "value":
+                if not (isinstance(value, dict) or isinstance(value, list)):
+                    yield value
+            else:
+                raise ValueError("'returned' keyword only accepts 'key' or 'value'.")
+            for ret in iterate_all(value, returned=returned):
+                yield ret
+    elif isinstance(iterable, list):
+        for el in iterable:
+            for ret in iterate_all(el, returned=returned):
+                yield ret
```

Comment:
There is special syntax sugar for that cases

```suggestion
            yield from iterate_all(el, returned=returned)
```